src/mlagility/api/ortmodel.py

```diff
@@ -63,23 +68,26 @@ def _execute(self, repetitions: int, backend: str = "local") -> MeasuredPerforma
             os.remove(self._ort_error_file)
 
         if backend == "remote":
-            devices.execute_ort_remotely(self.state, self.device, repetitions)
+            devices.execute_ort_remotely(
+                self.cache_dir, self.build_name, self.device_type, repetitions
+            )
         elif backend == "local":
-            devices.execute_ort_locally(self.state, self.device, repetitions)
+            devices.execute_ort_locally(
+                self.cache_dir, self.build_name, self.device_type, repetitions
+            )
         else:
             raise ValueError(
                 f"Only 'remote' and 'local' are supported, but received {backend}"
             )
 
         return MeasuredPerformance(
-            mean_latency=self._mean_latency,
-            throughput=self._throughput,
-            device=self._device,
-            device_type=self.device,
-            build_name=self.state.config.build_name,
+            mean_latency=self.mean_latency,
+            throughput=self.throughput,
+            device=self.device_name,
+            device_type=self.device_type,
+            build_name=self.build_name,
         )
 
 
 def load(build_name: str, cache_dir=build.DEFAULT_CACHE_DIR) -> ORTModel:
-    state = build.load_state(cache_dir=cache_dir, build_name=build_name)
-    return ORTModel(state)
+    return ORTModel(build_name=build_name, cache_dir=cache_dir)
```

Comment:
nit: This function has become pretty pointless 👀

Reply:
Removed pointless function